components/camel-jpa/src/test/java/org/apache/camel/component/jpa/JpaComponentTest.java

```diff
@@ -44,6 +44,7 @@ public void testJpaComponentConsumerHasLockModeType() throws Exception {
         JpaConsumer consumer = (JpaConsumer) jpa.createConsumer(null);
 
         assertEquals(LockModeType.PESSIMISTIC_WRITE, consumer.getLockModeType());
+        comp.close();
```

Comment:
What about using a try-with-resource statement instead?

Reply:
Code will be something like: 

```
@Test
public void testJpaComponentConsumerHasLockModeType() throws Exception {
    try (JpaComponent comp = new JpaComponent()) {
        comp.setCamelContext(context);
        assertNull(comp.getEntityManagerFactory());
        assertNull(comp.getTransactionManager());
        JpaEndpoint jpa
            = (JpaEndpoint) comp.createEndpoint("jpa://" + SendEmail.class.getName() + "?lockModeType=PESSIMISTIC_WRITE");
        JpaConsumer consumer = (JpaConsumer) jpa.createConsumer(null);
        assertEquals(LockModeType.PESSIMISTIC_WRITE, consumer.getLockModeType());
    } catch (Exception e) {
        throw e; //or e.printStackTrace();
    }
}
```

Once we will not handle the exception I believe is not necessary use try-with-resource.

Reply:
Why do you catch the exception ? You could simply throw it

Reply:
That is true. But don't is a bit confusing use try without catch?

Reply:
No because the try-with-resources statement has been created for this purpose. Read this https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html for more details.

Reply:
I know how to use this but I really find it a bit confusing. But no problem, I will do it.